### DISCUSSION TITLE
Trying to find the closest node from a weighted edge

### DISCUSSION THREAD

**Discussion by rhjennings** [graphology/graphology#425] [Q&A] [answered]
I've used graphs in the past but only in Java. I found this and  I think it will work for me, however since I'm newer to typescript, I'm stuck.

This is my code to add planets to the graph
`

    this.graph.addNode(homeWorld.id, homeWorld);
    planets.forEach(p => {
      this.graph.addNode(p.id, p);
      const distance: number = this.locationService.getDistance(
        this.buildLocationFromPlanet(homeWorld), this.buildLocationFromPlanet(p));
      this.graph.addEdge(homeWorld.id, p.id, {type: 'DISTANCE', weight: distance});
    });
`

next I have a function to find the closest planet based on the edge (DISTANCE)

`

    public findNearestOwnedPlanet(planet: Planet): Planet {
        const paths: SingleSourceDijkstraResult = dijkstra.singleSource(this.graph, planet.id, 'DISTANCE');
        return null;
    }

`

Ignoring the returning the null, I'm unsure how to extract the planet value from the paths (which is a {[key: string]: string[]}).
Or I might want to be getting something the incorrect way.

Any thoughts??

**Comment by Yomguithereal** [comment, marked as answer]
Hello @rhjennings. If you just need to find the closest planet (linked by an edge) to a specific planet, I am not sure you need to use the Dijkstra algorithm as it seems a bit overkill. You just need to find the planet's neighbor having the min distance basically, like so:

```js
const closestDistance = Infinity;
const closestPlanet = null;

graph.forEachEdge(targetPlanet, (edge, attr) => {
  if (attr.distance < closestDistance) {
    closestDistance = attr.distance;
    closestPlanet = graph.opposite(targetPlanet, edge);
  }
});

// Now to get the "value" of the closestPlanet, i.e. its attributes (beware, if your targetPlanet has no neighbors, closestPlanet will be null of course)
graph.getNodeAttributes(closestPlanet);
```

Does that work for you?

**Reply by rhjennings** [reply]
Sorry for the delay. I had to try several things to get the results I wanted. This helped a lot! So thanks for your help!
